src/allowlist/src/main.rs

```diff
@@ -331,3 +356,26 @@ where
         Ok(Self(connection, request_summary, parsed_address))
     }
 }
+
+// Asynchronous shutdown signal allowing CTRL+C or SIGTERM as a trigger.
+async fn shutdown_signal() -> Result<(), String> {
+    #[cfg(unix)]
+    let terminate_signal = async {
+        signal::unix::signal(signal::unix::SignalKind::terminate())
+            .map_err(|error| InitError::SIGTERMHandler(error).to_string())?
+            .recv()
+            .await;
+        Ok(())
+    };
+
+    #[cfg(not(unix))]
+    let terminate_signal = std::future::pending();
+
+    tokio::select! {
+        result = signal::ctrl_c() => {
+            result.map_err(|error| InitError::CtrlCHandler(error).to_string())?;
+            Ok(())
+        },
+        result = terminate_signal => { result }
+    }
+}
```

Comment:
I'm slightly confused here. Why is one for Unix and the other not for Unix ? Both SIGTERM and SIGINT are signals that can be sent to a program in a Unix environment, why not handle both ?

Reply:
This handler is implemented to comply with [AWS container best practices](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-considerations.html), namely

> Make your application handle `SIGTERM`

However, per the the [Microsoft developer docs](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-170):

> `SIGTERM` signals aren't generated under Windows

Hence the additional handling of `CTRL+C` in the [Axum graceful shutdown example](https://github.com/tokio-rs/axum/blob/179942d9cd739f45c4184c87c7976ae583e80014/examples/graceful-shutdown/src/main.rs#L51) I modeled this off of.

As to your question:

> Why is one for Unix and the other not for Unix?

This is to make it so that the await for `terminate_signal` never comes on a non-Unix system.

As to your second question:

> Both SIGTERM and SIGINT are signals that can be sent to a program in a Unix environment, why not handle both ?

Yes, theoretically, `SIGINT` could be sent to the container, but both AWS (see docs above) and Docker (see [Compose docs](https://docs.docker.com/compose/faq/#why-do-my-services-take-10-seconds-to-recreate-or-stop) ) send `SIGTERM` . I'd rather not clutter up the implementation since we don't want to interrupt the program, but rather, just stop it